velox/exec/SetAccumulator.h

```diff
@@ -86,6 +102,37 @@ struct SetAccumulator {
     }
   }
 
+  void readNullIndex(common::InputByteStream& stream) {
+    VELOX_CHECK(!nullIndex.has_value());
+    const auto streamNullIndex = stream.read<vector_size_t>();
+    if (streamNullIndex != kNoNullIndex) {
+      nullIndex = streamNullIndex;
+    }
+  }
+
+  inline bool isNullIndex(size_t i) {
+    return nullIndex.has_value() && i == nullIndex.value();
+  }
+
+  /// Deserializes accumulator from previously serialized value.
+  void deserialize(
+      const StringView& serialized,
+      HashStringAllocator* /*allocator*/) {
+    // The serialized value is the nullOffset (kNoNullIndex if no null is
+    // present) followed by the unique values ordered by index.
+    common::InputByteStream stream(serialized.data());
+    readNullIndex(stream);
+
+    size_t i = 0;
+    const auto size = serialized.size();
+    while (stream.offset() < size) {
```

Comment:
I wonder if it makes sense to include the total number of entries in the serialized blob and make sure we read back exactly that many.

Reply:
Thats a good idea. Would make the logic more robust.